### ISSUE TITLE
Multiple ContinueJobWith failing to process in 1.7.x

### ISSUE THREAD

**Issue by psiclopz** [HangfireIO/Hangfire#1913]
We are on 1.6.30 currently, recently upgraded from 1.6.21, but wanted to upgrade to 1.7.x.  We also use MSSQL storage. When attempting to upgrade to 1.7.x, ContinueJobWith (previously ContinueWith) call chains fail to process.

Inside the initial job we have this:

                                      step1.CreateBatch();

					var step2JobId = BackgroundJob.ContinueJobWith(performContext.BackgroundJob.Id, () =>
						Execute_Step2(transactionKey, performContext));

					var step3JobId = BackgroundJob.ContinueJobWith(step2JobId, () =>
						Execute_Step3(transactionKey, performContext));

					var step4JobId = BackgroundJob.ContinueJobWith(step3JobId, () =>
						Execute_Step4(transactionKey, performContext));

					var step5JobId = BackgroundJob.ContinueJobWith(step4JobId, () =>
						Execute_Step5(transactionKey, performContext));

					BackgroundJob.ContinueJobWith(step5JobId, () =>
						Execute_Step6(transactionKey, performContext));
`
The main job processes and completes, and the first ContinueJobWith call, for Step 2, queues, and successfully completes, and the second ContinueJobWith call, for Step 3, queues, and then never processes.  Looking at the hangfire dashboard shows the correct parent ID, the id for step 2's job, and going to that job shows that it has a success state.

The only thing that has changed is upgrading to 1.7.x, and I have tried ~ 5 different versions, from 1.7.1 all the way to 1.7.24.

There is also nothing in the logs at all.  And other jobs process as normal, just the > 1 ContinueJobWith chains.  

If there is any other information I can provide to help troubleshoot this, please let me know.  Because we would really like to upgrade to 1.7.x.

**Comment by odinserj** [comment]
Can you show some dashboard screenshots that shows the issue – for example continuation job details page that fails to run and its antecedent job details page?

**Comment by psiclopz** [comment]
Queued job that fails to run
![QueuedJob](https://user-images.githubusercontent.com/3727194/127697401-dcdaeff3-50a0-4588-a005-8fec35dc6f50.png)

Antecedent job
![SucceededParent](https://user-images.githubusercontent.com/3727194/127697405-259b111c-718a-4b21-8c85-966d636c2d2f.png)

**Comment by odinserj** [comment]
Thanks for the screenshots, I see that continuation job was actually enqueued (so continuation feature works fine) to the `clearinghouse_submissions_...` queue. So now we should check that the queue is listened by some server, so please also send me screenshot of the `Servers` page to ensure that.

**Comment by psiclopz** [comment]
Here you are.  And I'll note that this has worked for years.  No code has changed at all.  The only thing that changed was upgrading from 1.6.x -> 1.7.x
![Servers](https://user-images.githubusercontent.com/3727194/127707946-c9756765-e560-488d-a3a9-c21cf0bfb5d3.png)

**Comment by psiclopz** [comment]
If I click on the queue name here
![QueuedJobCloseup](https://user-images.githubusercontent.com/3727194/127708501-9b438caf-e229-4e32-9dc7-0a89bb66c6ca.png)

it takes me to this screen that says the queue is empty.

![EmptyQueue](https://user-images.githubusercontent.com/3727194/127708575-f7f1f858-81aa-43ad-8a21-90fa978d0e45.png)

**Comment by psiclopz** [comment]
Ah, I think I just figured out the issue.  It looks like years ago, when HF was added to our product, some engineers manually upped the size of the JobQueue.Queue and Server.Id columns to 200 chars from 20/50 to support longer names.  And apparently 1.6.x didn't mind.  But now the inserts for 1.7 are truncating the queue name when it goes into the table to 50 chars, which is why it's never getting processed.  So it's less a problem with the Continues, than a problem with that specific queue name.  I'll change the queue names to drop them below 50 characters and I bet that will fix the issue.  I will update shortly.

On a related note, the queue field should support longer names :-)
